### PR TITLE
More planet/entity class method migrations

### PR DESCRIPTION
**Pull request by mrcage**
**Merge planet_fleet_resource_interaction branch first before this one**

This change includes a refactoring of the crypto center related code as it is heavily connected to coordinates and resources of planets.

### CODE REVIEW

**Comment by glaubinix** [comment]
Looks like there is still something off with the build

**Comment by mrcage** [comment]
Fixed db test issues

File: htdocs/content/shipyard.php [hunk 1]
```diff
@@ -654,44 +633,43 @@
...
+                        [b]" . RES_FUEL . ":[/b] " . nf($bc['fuel']) . "
+                        [b]" . RES_FOOD . ":[/b] " . nf($bc['food']) . "
 
                         [b]Rohstoffe auf dem Planeten[/b]
-                        [b]".RES_METAL.":[/b] ".nf($planet->resMetal)."
-                        [b]".RES_CRYSTAL.":[/b] ".nf($planet->resCrystal)."
-                        [b]".RES_PLASTIC.":[/b] ".nf($planet->resPlastic)."
-                        [b]".RES_FUEL.":[/b] ".nf($planet->resFuel)."
-                        [b]".RES_FOOD.":[/b] ".nf($planet->resFood)."";
+                        [b]" . RES_METAL . ":[/b] " . nf($planet->resMetal - $totalMetal) . "
```
Comment (glaubinix) [inline htdocs/content/shipyard.php:663]: Could also fetch the planet after we change any data to make sure it is up to date

Comment (glaubinix) [reply htdocs/content/shipyard.php:663]: As for later, ideally every such request would be POST/PUT/DELET and then redirect to e.g. the shipyard. This way the page that renders will always have data that is up to date.

**Pull request merged**